### DISCUSSION TITLE
Multi-Element Airfoil Optimization

### DISCUSSION THREAD

**Discussion by mlau1424** [mdolab/dafoam#82] [General]
Hello,

I am starting a project involving optimizing a multi-element airfoil. I also should state I am new to CFD analysis, so thank you in advance for your patience. I have started to experiment with the tutorial cases, but could not find any support about multiple elements within the mesh. 

My main question is, before delving in too deep, will DAFoam have the capability to optimize a multi-element airfoil? Initially, the dimensions of the elements that make up the airfoil can all be equal, but further optimization of each element would also be useful.

Any guidance or recommendations would be greatly appreciated!

Thank you.

**Comment by friedenhe** [comment]
Yes, it can do multi-element airfoil optimization with DAFoam. You need to set up a multi-block FFD file (check [this script](https://github.com/DAFoam/tutorials/blob/master/DPW4_Aircraft/FFD/genFFD.py) on how to generate multiblock FFDs, you can use Paraview load a [multi-block FFD file](https://github.com/DAFoam/tutorials/blob/master/DPW4_Aircraft/FFD/wingTailFFD.xyz)). In this FFD file, each block will cover one airfoil element. You can then change these two FFD blocks in the optimization. Refer to this [run script](https://github.com/DAFoam/tutorials/blob/master/DPW4_Aircraft/runScript2FFDs.py) for optimizations with multiblock FFDs.

**Comment by friedenhe** [comment]
@mlau1424 In case you are still interested, we have recently added a [tutorial](https://dafoam.github.io/mydoc_tutorials_aero_30n30p.html) for multi-element airfoil optimization.

**Reply by mlau1424** [reply]
@friedenhe This is great, thank you for the update, it is greatly appreciated.
